frontend/apps/reader/modules/readeruserhyph.lua

```diff
@@ -0,0 +1,372 @@
+        return
+    end
+    self.new_dict = io.open(self.new_dict_file, "w")
+    if not self.dict then
+        logger.err("UserHyph: could not open " .. self.new_dict_file)
+        return
+    end
+    return true
+end
+
+-- reads the first dictionary entries from the old dictionary
+-- and write them to the new dict, as long the entry is alphabetically lower than word
+-- if the word matches the dictionary entry, the hypenation part is formatted and returned
+function ReaderUserHyph:findEntry(word)
+    -- scan hyphenation dictionary for selected word
+    local word_lower = Utf8:lower(word)
+    self.next_line = self.dict:read()
+    while self.next_line and Utf8:lower(self.next_line:sub(1, self.next_line:find(";") - 1)) < word_lower do
+        self.new_dict:write(self.next_line .. "\n")
+        self.next_line = self.dict:read()
+    end
+
+    local hyphenation
+    -- check if a hyphenation is found for word
+    if self.next_line and Utf8:lower(self.next_line:sub(1, self.next_line:find(";") - 1)) == word_lower then
+        hyphenation = string.sub(self.next_line, self.next_line:find(";") + 1) -- hyphenation found
+        self.next_line = nil -- Important for not duplicating the entry in the dictionary
+    end
+    return hyphenation
+end
+
+-- writes on entry to new dictionary
+function ReaderUserHyph:writeEntry(line)
+    self.new_dict:write(Utf8:lower(line) .. "\n")
+end
+
+-- reads the rest of the old dictionary and writes this to the reset
+-- if self.new_dict~=nil, an additional line can be written
+function ReaderUserHyph:writeRest()
+    -- write old entry if there was one
+    if self.next_line then
+        self.new_dict:write(self.next_line .. "\n")
+    end
+    -- copy rest of file
+    repeat
+        self.next_line = self.ui.userhyph.dict:read()
+        if self.next_line then
+            self.new_dict:write(self.ui.userhyph.next_line .. "\n")
+        end
+    until (not self.next_line)
+end
+
+-- closes all open files and invalidates variables
+-- mode
+--    KEEP_ORIGINAL .. keep the original dict
+--    USE_NEW .. use the new dict
+function ReaderUserHyph:closeDictionary(mode)
+    if not mode then
+        logger.err("ReaderUserHyp: BUG ond close Dictionary")
+        return
+    end
+
+    self.ui.userhyph.dict:close()
+    self.ui.userhyph.new_dict:close()
+    if mode == 1 then
+        os.remove(self.ui.userhyph.new_dict_file)
+    elseif mode == 2 then
+        os.remove(self.ui.userhyph.dict_file)
+        os.rename(self.ui.userhyph.new_dict_file, self.ui.userhyph.dict_file)
+    end
+
+    self.next_line = nil
+    self.dict_file = nil
+    self.new_dict_file = nil
+    self.dict = nil
+end
+
+function ReaderUserHyph:modifyUserEntry(word)
+    if word:find("[ ,;-%.]") then return end -- no button if more than one word
+
+    self:openDictionary()
+    local suggested_hyphenation = self:findEntry(word)
+
+    if self.ui.document and self.ui.document.getHyphenation then
```

Comment:
readeruserhyph is only loaded with CRE documents, so you'll always have self.ui.document.getHyphenation true.